### ISSUE TITLE
Import from Facebook?

### ISSUE THREAD

**Issue by crashroach** [friendica/friendica#5212]
The README says you can post to Facebook, but is there a way to import all your old posts from Facebook? It should be possible since Facebook now lets you download "all" your data. That would improve usability and ease of transition for the average person immensely.

**Comment by MrPetovan** [comment]
Thanks @crashroach for the suggestion, this seems feasible, although I don't know the exact extent. I can't access my Facebook account right now, which means I can't get a copy of my data or verify the format it's using.

**Comment by crashroach** [comment]
According to my research, the timeline is stored as one big html file, and the images are stored separately. 
Also there's the graph api, which supposedly lets you get posts directly from facebook, but I've never done anything server side and I don't know php (though there is also a js api), so I'm not sure if it will allow just pulling all a user's content directly.
https://developers.facebook.com/docs/php/api/5.0.0

**Comment by AndyHee** [comment]
It think this would be valuable feature, especially in terms of photos. Plus it could be serious crowd-puller for waving FB leavers.

I looked at my saved FB archive and the structure looks quite simple.

**Comment by AndyHee** [comment]
*Archive Structure**
video/                   //all videos and vdo screenshots
photos/                 // separate folders that contain each an index.htm file and photos. Profile photos is the root of this folder
html/                     // all htm files, such as timeline, events, friends, photos, contact_info
index.htm


In a way such an archive could even facilitate the Friendica registration process, as some personal information could be pull out and used to populate a new users profile.

**Comment by MrPetovan** [comment]
The only thing we can’t import in this list is relationships.

**Comment by AndyHee** [comment]
Yes, you only get their _names_.

**Comment by AndyHee** [comment]
Ohh. This is not true. I found and whole address book in there that contains toons of people with their phone numbers and email addresses. I think that came from was some import feature I used once by connecting gmail and FB. Scary!

**Comment by MrPetovan** [comment]
And that’s exactly why we won’t *use* it.

**Comment by AndyHee** [comment]
Yes, I fully agree. We need to make sure this would be properly GDPR compliant and give very detailed explanations what happens during and after the import process with the data.


Do you think such feature would be best in a separate addon?

**Comment by MrPetovan** [comment]
Yes, not every node will be Facebook-friendly, so it should be up to admins to decide whether to enable this import or not.

**Comment by AndyHee** [comment]
Plus uploading massive data files (0.5GB?) onto a node might not be feasible for smaller nodes.

Sorry to bother you, Hypolite. Could you change the title once more: 
"New Addon suggestion: Import data from Facebook personal archive"

Thanks!

**Comment by MrPetovan** [comment]
Yeah, I'm not sure how we're going to handle that. Suggest to host the file in a public place in order for the node to download it seems inconvenient as well.

**Comment by AndyHee** [comment]
Possibly a capping limit set by the admin,  50% of the size of the /tmp/ directory.

My own FB archive file is 50MB after ~10 years of moderate usage. But I have heard some people have accumulated much more.

**Comment by AndyHee** [comment]
Actually the size of the archive can be easily controlled nowadays.
Also they just introduced an additional format: JSON

> You can download a copy of your Facebook information at any time. You can download all of it at once, or you can select only the types of information and date ranges you want. You can choose to receive your information in an HTML format that is easy to view, or a JSON format, which could allow 
another service to more easily import it.

**Comment by tobiasd** [comment]
While FB import would be nice, fixung #1398 on the way there would be nicer ;-)

**Comment by zleap** [comment]
I agree with this idea  but also agree with @tobiasd  about issue #1398 as being able to export / import data from / to friendi.ca nodes is also really useful.

**Comment by tobiasd** [comment]
> export

This is already working. ATM only the import of the profile base data is working though.

**Comment by rabuzarus** [comment]
> export

Yes, but this can be improved - e.g. also exporting events or photos

**Comment by tobiasd** [comment]
> Yes, but this can be improved

Absolutely :D

**Comment by twinklebob** [comment]
I'm currently considering leaving Facebook completely. If there was a way I could take my history of photos, videos and posts, that would be massive.

**Comment by MrPetovan** [comment]
I joined Friendica because my Facebook account was frozen so I don’t have access to a Facebook archive to work on this feature, sorry. 😕

**Comment by twinklebob** [comment]
How much of a task do you think it would be? I'm a PHP developer and I still have a facebook account, so I might be able to have a go at this. However I don't know much about the friendica codebase.

**Comment by MrPetovan** [comment]
The main thing that I'm missing to work on this is the structure of the Facebook archive. All I would need from you would be a detailed documentation of the format. A question off the top of my head: are there different types of archives? With/without images/videos? Multimedia archive might prove harder to process because of their sheer size so we might focus on posts-only first.

**Comment by bkil** [comment]
* https://nequalsonelifestyle.com/2021/12/07/kyanite-beta2-updates/
* https://gitlab.com/HankG/mysocialportal/-/releases/kyanite_v0.1.2
> Kyanite A Flutter-based cross platform desktop application for viewing the Facebook account archive that a user can request from the system. This takes the form of a zip file that can be unzipped on your desktop system and have this app point to it.
> _(Dart, MPL, export parser, backup explorer)_

* https://gitlab.com/HankG/mysocialportal
> MySocialPortal: A library and application ecosystem for unifying your social media and social graph data from many networks into a common view.

**Comment by MrPetovan** [comment]
Thank you for the links but I still need a source dataset, either real or randomized like @HankG (hi! 👋) used for his screenshot.

**Comment by NotFluffy** [comment]
Has there been work on this? Been manually moving my posts from facebook and would love to be able to automate it...

**Comment by MrPetovan** [comment]
Unfortunately not. I've not had access to my still existing Facebook account since 2016 so my willingness to work on this task has weaned. 😔

**Comment by NotFluffy** [comment]
I totaly understand that! 
I found a roundabout way to go at it by transfering the posts to wordpress, I'll figure it out from there :)

**Comment by Helianthropy** [comment]
I’ve started a file to identify the structure of the archive. My layout may be a bit unorthodox, but I think it does a good job of revealing the depth of Facebook’s information collection.

Obviously, some of this stuff may not be useful, so would like to identify the most useful parts for a sample archive (obviously, the personal content like posts and photos make sense), but having never used Friendica (I know, I know—I do want to), I’m not sure of what all would be useful to have.

Additionally, I didn’t make use of all of Facebook’s features, so may have holes in my archive (I indicated having never created stories, for instance). Please use this file as a jumping off point to identify missing files from your own archives and/or indicate which files would be most useful to have in a sample archive.

[facebook-archive-structure.txt](https://github.com/user-attachments/files/23960158/facebook-archive-structure.txt)
